syntax = "proto3";

package xpilot;

import "google/protobuf/timestamp.proto";
import "xplanedata.proto";
import "pluginhash.proto";
import "pluginversion.proto";
import "cslvalidate.proto";
import "addplane.proto";
import "changemodel.proto";
import "positionupdate.proto";
import "removeplane.proto";
import "removeallplanes.proto";
import "networkconnected.proto";
import "networkdisconnected.proto";
import "airplaneconfig.proto";
import "nearbycontrollers.proto";
import "clearnearbycontrollers.proto";
import "privatemessagereceived.proto";
import "privatemessagesent.proto";

message Wrapper {
	google.protobuf.Timestamp timestamp = 1;
	oneof msg {
		XplaneData xplane_data = 2;
		PluginHash plugin_hash = 3;
		PluginVersion plugin_version = 4;
		CslValidate csl_validate = 5;
		AddPlane add_plane = 6;
		ChangeModel change_model = 7;
		PositionUpdate position_update = 8;
		RemovePlane remove_plane = 9;
		RemoveAllPlanes remove_all_planes = 10;
		NetworkConnected network_connected = 11;
		NetworkDisconnected network_disconnected = 12;
		AirplaneConfig airplane_config = 13;
		NearbyControllers nearby_controllers = 14;
		ClearNearbyControllers clear_nearby_controllers = 15;
		PrivateMessageReceived private_message_received = 16;
		PrivateMessageSent private_message_sent = 17;
	}
}